scopes/component/ui/compare/aspects/compare-aspects.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { HTMLAttributes, useState, useMemo } from 'react';
+import classNames from 'classnames';
+import { gql } from '@apollo/client';
+import { isEqual } from 'lodash';
+import { useDataQuery } from '@teambit/ui-foundation.ui.hooks.use-data-query';
+import { HoverSplitter } from '@teambit/base-ui.surfaces.split-pane.hover-splitter';
+import { Collapser } from '@teambit/ui-foundation.ui.buttons.collapser';
+import { SplitPane, Pane, Layout } from '@teambit/base-ui.surfaces.split-pane.split-pane';
+import { useIsMobile } from '@teambit/ui-foundation.ui.hooks.use-is-mobile';
+import { RoundLoader } from '@teambit/design.ui.round-loader';
+import { CompareStatus, CompareStatusResolver, useComponentCompareContext, useComponentCompareParams } from '@teambit/component.ui.compare';
+import { CodeCompareTree } from '@teambit/code.ui.code-compare';
+import { ComponentCompareAspectsContext, useComponentCompareAspectsContext } from './compare-aspects-context';
+import { CompareAspectView } from './compare-aspect-view';
+
+import styles from './compare-aspects.module.scss';
+
+export type ComponentCompareAspectsProps = { host: string } & HTMLAttributes<HTMLDivElement>;
+export type ComponentAspectData = {
+  icon?: string;
+  name?: string;
+  config: any;
+  data: any;
+  aspectId: string;
+};
+
+const GET_COMPONENT_ASPECT_DATA = gql`
+  query GetComponentAspectData($id: String!, $extensionId: String!) {
+    getHost(id: $extensionId) {
+      id # used for GQL caching
+      get(id: $id) {
+        id {
+          name
+          version
+          scope
+        }
+        aspects {
+          aspectId: id
+          config
+          data
+          icon
+        }
+      }
+    }
+  }
+`;
+
+export function ComponentCompareAspects({ host, className }: ComponentCompareAspectsProps) {
+  const componentCompareContext = useComponentCompareContext();
+  const base = componentCompareContext?.base;
+  const compare = componentCompareContext?.compare;
+
+  const isCompareVersionWorkspace = componentCompareContext?.isCompareVersionWorkspace;
+
+  const baseId = `${base?.id.fullName}@${base?.id.version}`;
+  const compareId = isCompareVersionWorkspace ? compare?.id.fullName : `${compare?.id.fullName}@${compare?.id.version}`;
+
+  const { data: baseAspectData, loading: baseLoading } = useDataQuery(GET_COMPONENT_ASPECT_DATA, {
+    variables: { id: baseId, extensionId: host },
+    skip: !base?.id,
+  });
+
+  const { data: compareAspectData, loading: compareLoading } = useDataQuery(GET_COMPONENT_ASPECT_DATA, {
+    variables: { id: compareId, extensionId: host },
+    skip: !compare?.id,
+  });
+
+  const loading = baseLoading || compareLoading || componentCompareContext?.loading;
+  const baseAspectList: ComponentAspectData[] = baseAspectData?.getHost?.get?.aspects || [];
+  const compareAspectList: ComponentAspectData[] = compareAspectData?.getHost?.get?.aspects || [];
```

Comment:
maybe extract to a hook?
useComponentAspectCompare(componentId) => ComponentAspectCompareModel